src/imperative.rkt

```diff
@@ -244,7 +244,8 @@
        [(and (or (equal? (language-name (*lang*)) "c")  ; Only C and Scala need to emit a temporary variable
                  (equal? (language-name (*lang*)) "scala"))
              (or (not (equal? curr-prec new-prec))
-                 (not (equal? curr-round new-round))))
+                 (not (equal? curr-round new-round)))
+             (list? body))                            ; except if it is a no op
```

Comment:
This change does not fix #86. Consider the following core:
```
(FPCore simple (x)
  :precision binary64
  (! :precision binary32 (! :precision binary32 x)))
```
The result is the same as before:
```c
double ex0(double x) {
	float tmp = x;
	return (double) tmp;
}
```

Reply:
Ahh thanks. I guess this one is a little more complicated.

Reply:
I think the following idea will work here. `convert-expr` should return both the result expression (as a string) and its type (precision). Then types of all temporary variables are defined by types returned by `convert-expr`.

Reply:
That's actually what the `smt` compiler does. The simpler solution would be to also check for `!` in the body of the expression.

Reply:
Now that I think about it, constants need to be rounded, so a temporary variable should be emitted unless the body contains another annotation or a variable.

Reply:
It is hard to check all possible cases. For example:
```
(FPCore simple (x)
  :precision binary64
  (! :precision binary32 (let ((y x)) y)))
```
So it would be better to make changes similar to the `smt` compiler. Then it will also be possible to reuse `imperative` functions in the `smt` compiler (`convert-application` should be generalized to do this).

Reply:
That's a good counterexample. Ok, I'll try to get this fixed quickly.